chainercv/datasets/camvid/camvid_dataset.py

```diff
@@ -0,0 +1,80 @@
+import glob
+import os.path as osp
+import shutil
+
+import numpy as np
+
+import chainer
+from chainer.dataset import download
+from chainercv import utils
+from chainercv.utils import read_image
+
+root = 'pfnet/chainercv/camvid'
+url = 'https://github.com/alexgkendall/SegNet-Tutorial/archive/master.zip'
+
+
+def get_camvid():
+    data_root = download.get_dataset_directory(root)
+    download_file_path = utils.cached_download(url)
+    if len(glob.glob(osp.join(data_root, '*'))) != 9:
+        utils.extractall(download_file_path, data_root, osp.splitext(url)[1])
+    data_dir = osp.join(data_root, 'SegNet-Tutorial-master/CamVid')
+    if osp.exists(data_dir):
+        for fn in glob.glob(osp.join(data_dir, '*')):
+            shutil.move(fn, osp.join(data_root, osp.basename(fn)))
+        shutil.rmtree(osp.dirname(data_dir))
+    return data_root
+
+
+class CamVidDataset(chainer.dataset.DatasetMixin):
+
+    """Dataset class for a semantic segmantion task on CamVid `u`_.
+
+    .. _`u`: https://github.com/alexgkendall/SegNet-Tutorial/tree/master/CamVid
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/camvid`.
+        mode ({'train', 'val', 'test'}): Select from dataset splits used
+            in VOC.
+
+    """
+
+    def __init__(self, data_dir='auto', mode='train'):
+        if mode not in ['train', 'val', 'test']:
+            raise ValueError(
+                'Please pick mode from \'train\', \'val\', \'test\'')
+
+        if data_dir == 'auto':
+            data_dir = get_camvid()
+
+        img_list_fn = osp.join(data_dir, '{}.txt'.format(mode))
```

Comment:
Actually, fn is used in other part of ChainerCV (e.g. `datasets/cub/cub_utils`).